src/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -68,12 +69,13 @@ def __init__(self) -> None:
         self.contexts: dict[int, Context] = {}
         self.task_ins_store: dict[UUID, TaskIns] = {}
         self.task_res_store: dict[UUID, TaskRes] = {}
+        self.task_ins_id_to_task_res: dict[UUID, TaskRes] = {}
 
         self.node_public_keys: set[bytes] = set()
         self.server_public_key: Optional[bytes] = None
         self.server_private_key: Optional[bytes] = None
 
-        self.lock = threading.Lock()
+        self.lock = threading.RLock()
```

Comment:
Just for my understanding, why did you switch to `RLock`?

Reply:
Oh, I implemented a `_force_delete_tasks_by_ids` method, in which the lock is also used. Switching to `RLock` allows the `get_task_res` to directly invoke `_force_delete_tasks_by_ids` without having to release the lock first, as long as they are in the same thread.